pkg/kubectl/resource/builder.go

```diff
@@ -166,16 +171,42 @@ func (b *Builder) FilenameParam(enforceNamespace bool, filenameOptions *Filename
 }
 
 // Unstructured updates the builder so that it will request and send unstructured
-// objects by default. Calling this method resets Local().
+// objects. Unstructured objects preserve all fields sent by the server in a map format
+// based on the object's JSON structure which means no data is lost when the client
+// reads and then writes an object. Use this mode in preference to Internal unless you
+// are working with Go types directly.
 func (b *Builder) Unstructured() *Builder {
 	if b.unstructured == nil {
-		b.errs = append(b.errs, fmt.Errorf("no unstructured builder provided"))
+		b.errs = append(b.errs, fmt.Errorf("no unstructured mapper provided"))
+		return b
+	}
+	if b.mapper != nil {
+		b.errs = append(b.errs, fmt.Errorf("another mapper was already selected, cannot use unstructured types"))
 		return b
 	}
 	b.mapper = b.unstructured
 	return b
 }
 
+// Internal updates the builder so that it will convert objects off the wire
+// into the internal form if necessary. Using internal types is lossy - fields added
+// to the server will not be seen by the client code and may result in failure. Only
+// use this mode when working offline, or when generating patches to send to the server.
+// TODO: Introduce a Versioned() form that performs this same function, but biases
```

Comment:
The warning on this method would remain identical.  For a given snapshot of external types, the internal ones are just as good.